### ISSUE TITLE
wget command to download all files in a dataset not preserving file hierarchy

### ISSUE THREAD

**Issue by meghangoodchild** [IQSS/dataverse#8836]
**What steps does it take to reproduce the issue?**

For Dataverse installation that is using S3 storage.
When using the wget command (example pasted below) for a dataset with a file structure hierarchy (i.e., tree structure):

`wget -r -e robots=off -nH --cut-dirs=3 --header "X-Dataverse-key: $API_TOKEN" --content-disposition https://borealisdata.ca/api/datasets/:persistentId/dirindex?persistentId=IDENTIFIER"`

* When does this issue occur?

Every time the command is used on a dataset with a tree structure

* Which page(s) does it occurs on?

N/A

* What happens?

It downloads one directory “dirindex” and one directory with a name based on the dataset DOI. The latter contains a flat structure of the dataset files without retaining the file hierarchy of the dataset. If there are files contained in different folders that have the same name, they will get overwritten in the destination folder. 

* To whom does it occur (all users, curators, superusers)?

All users

* What did you expect to happen?

I expected that the dataset should download in a file hierarchy as outlined in the documentation, “Using this API, `wget --recursive` (or a similar crawling client) can be used to download all the files in a dataset, preserving the file names and folder structure; without having to use the download-as-zip API.”  [](https://guides.dataverse.org/en/latest/api/native-api.html#view-dataset-files-and-folders-as-a-directory-index)

**Which version of Dataverse are you using?**

v.5.8.3

**Any related open or closed issues to this bug report?**

None found.

**Other notes**
 Based on our testing, we are wondering if this is related to the file system. When using a local machine with a standard file system, the command seems to work. Borealis uses S3 emulation.

**Comment by pdurbin** [comment]
@meghangoodchild thanks for testing on both S3 and a local filesystem.

**Comment by pdurbin** [comment]
@shlake reported this as well: https://dataverse.zulipchat.com/#narrow/stream/378866-troubleshooting/topic/View.20Dataset.20files.20and.20folders.20as.20directory.20index/near/453510488

**Comment by shlake** [comment]
Yes, does not work on our S3 direct download system and demo.dataverse (which I think is not S3)?
